frontend/components/Archives.tsx

```diff
@@ -42,186 +39,232 @@ export interface IFetchArchives {
     startFrom: number,
     amount: number,
     setArchives: React.Dispatch<React.SetStateAction<IArchive[]>>,
-    setLoadedIndices: React.Dispatch<React.SetStateAction<number>>,
+    setLastLoadedIndex: React.Dispatch<React.SetStateAction<number>>,
     ownerProfiles: { [key: string]: any },
     setOwnerProfiles: React.Dispatch<
       React.SetStateAction<{ [key: string]: any }>
     >,
   ): Promise<void>;
 }
 
+export interface IFetchArchivesCount {
+  (
+    setArchivesCount: React.Dispatch<React.SetStateAction<number | undefined>>,
+  ): Promise<void>;
+}
+
 interface IOwners {
   [key: string]: LSP3ProfileMetadata;
 }
 
 interface ArchivesProps {
   fetchArchives: IFetchArchives;
+  fetchArchivesCount: IFetchArchivesCount;
 }
 
-const Archives: React.FC<ArchivesProps> = ({ fetchArchives }) => {
+const Archives: React.FC<ArchivesProps> = ({
+  fetchArchivesCount,
+  fetchArchives,
+}) => {
   const walletContext = useContext(WalletContext);
   const { networkConfig, provider, refineEventCounter } = walletContext;
+  const [archivesCount, setArchivesCount] = useState<number>();
   const [archives, setArchives] = useState<IArchive[]>();
   const [startIndex, setStartIndex] = useState(0);
-  const [loadedIndices, setLoadedIndices] = useState<number>(0);
+  const [slideAmount, setSlideAmount] = useState<number>(0);
+  const [lastLoadedIndex, setLastLoadedIndex] = useState<number>(0);
   const [ownerProfiles, setOwnerProfiles] = useState<IOwners>({});
+  const carouselRef = useRef<HTMLDivElement>(null);
+  const archiveContainerWidth = 130;
+  const archiveContainerGap = 24;
+
+  useEffect(() => {
+    const resizeObserver = new ResizeObserver((entries) => {
+      for (let entry of entries) {
+        setSlideAmount(
+          Math.floor(
+            entry.contentRect.width /
+              (archiveContainerWidth + archiveContainerGap),
+          ),
+        );
+      }
+    });
 
-  // Use the useBreakpointValue hook to determine the number of images to slide
-  const responsiveSlideValues = { base: 3, md: 5 };
-  const slideAmount = useBreakpointValue(responsiveSlideValues) || 5;
+    if (carouselRef.current) {
+      resizeObserver.observe(carouselRef.current);
+    }
 
-  const burntPixArchives = BurntPixArchives__factory.connect(
-    networkConfig.burntPixArchivesAddress,
-    provider,
-  );
+    return () => {
+      if (carouselRef.current) {
+        resizeObserver.unobserve(carouselRef.current);
+      }
+    };
+  }, [carouselRef]);
 
   useEffect(() => {
-    // Use the passed-in fetchArchives function with necessary arguments
-    fetchArchives(
-      startIndex,
-      slideAmount,
-      setArchives as Dispatch<SetStateAction<IArchive[]>>,
-      setLoadedIndices,
-      ownerProfiles,
-      setOwnerProfiles,
-    );
-  }, [fetchArchives, slideAmount, refineEventCounter]);
+    fetchArchivesCount(setArchivesCount);
+  }, [fetchArchivesCount, refineEventCounter]);
+
+  useEffect(() => {
+    slideAmount &&
```

Comment:
is this conditional required?

Reply:
yes because initially slide amount is 0 until page determines correct value. so when it's 0 doesnt make sense to fetch anything because we dont know how much to fetch